### PR TITLE
chore: WDS color algo comments

### LINKED ISSUE #24461: chore: WDS, add comments to `DarkModeTheme` and `LightModeTheme`

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Noting down color generation logic

Fixes #24461 

## Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/packages/design-system/theming/src/color/LightModeTheme.ts [hunk 1]
```diff
@@ -215,14 +232,16 @@ export class LightModeTheme implements ColorModeTheme {
     return color;
   }
 
-  // used only for generating child colors, not used as a token
+  // Subtle variant of bgAccent. Lighter and less saturated.
```
Comment (ichik) [inline app/client/packages/design-system/theming/src/color/LightModeTheme.ts:235]: We're going to use it eventually and we already have a few other tokens that are not being used by anything yet.

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts [hunk 2]
```diff
@@ -345,13 +365,16 @@ export class DarkModeTheme implements ColorModeTheme {
     return this.fg.clone();
   }
 
+  // Positive background, green.
   private get bgPositive() {
     const color = this.bgAccent.clone();
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:370]: Similarly with the negative, can we just write here new Color("green"),

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts [hunk 3]
```diff
@@ -367,9 +390,13 @@ export class DarkModeTheme implements ColorModeTheme {
   /*
    * Foreground colors
    */
+  // Main application foreground color.
+  // Applies to static text and similar. In light mode it is extremely dark (and therefore desatured) shade of user-set seed color.
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:394]: light mode » dark mode?

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts [hunk 4]
```diff
@@ -166,16 +177,19 @@ export class DarkModeTheme implements ColorModeTheme {
...
 
+    // For very light seeds it becomes impossible to produce hover state that is sufficiently perceptibly lighter, therefore we're switching to having hovers darker.
     if (this.seedLightness >= 0.85) {
       color.oklch.l = this.bgAccent.oklch.l - 0.07;
     }
 
     return color;
   }
 
+  // Active state of bgAccent. Slightly darker than the resting state to produce the effect of moving further from the viewer / being pushed down.
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:188]: If you decide to fix it, then you should not forget to fix it in other places too

**Review by KelvinOm — DISMISSED** [review]
Thanks for that! Code became much clearer.

**Review by KelvinOm — APPROVED** [review]

File: app/client/packages/design-system/theming/src/color/LightModeTheme.ts [hunk 5]
```diff
@@ -51,12 +51,12 @@ export class LightModeTheme implements ColorModeTheme {
       bgAccentSubtleHover: this.bgAccentSubtleHover.toString(),
       bgAccentSubtleActive: this.bgAccentSubtleActive.toString(),
       bgAssistive: this.bgAssistive.toString(),
-      bgPositive: this.bgPositive.toString(),
+      bgPositive: this.bgPositive.to("sRGB").toString(),
```
Comment (jsartisan) [inline app/client/packages/design-system/theming/src/color/LightModeTheme.ts:54]: Why is this needed? 🤔

File: app/client/packages/design-system/theming/src/color/LightModeTheme.ts [hunk 6]
```diff
@@ -112,6 +117,7 @@ export class LightModeTheme implements ColorModeTheme {
       color.oklch.l = 0.985;
     }
 
+    // Cold colors can have a bit more chroma while staying perceptually neutral
```
Comment (jsartisan) [inline app/client/packages/design-system/theming/src/color/LightModeTheme.ts:120]: What exactly cold colors are? blueish range you mean?

File: app/client/packages/design-system/theming/src/color/DarkModeTheme.ts [hunk 7]
```diff
@@ -195,19 +209,22 @@ export class DarkModeTheme implements ColorModeTheme {
...
     }
 
+    // If the color is too dark it won't be visible against bg, so it needs a minimum too.
     if (this.seedLightness < 0.2) {
       color.oklch.l = 0.2;
     }
 
-    if (this.seedChroma > 0.112 && !this.seedIsAchromatic) {
+    // If chroma is above the maximum, set it to maximum.
+    if (this.seedChroma > 0.112) {
```
[the remark this answers is not in the document]
Comment (jsartisan) [reply app/client/packages/design-system/theming/src/color/DarkModeTheme.ts:225]: +1

File: app/client/packages/design-system/theming/src/color/LightModeTheme.ts (hunk 5 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/color/LightModeTheme.ts:54]: Otherwise we output CSS with `oklch()` which may have unintended side-effects.

File: app/client/packages/design-system/theming/src/color/LightModeTheme.ts (hunk 6 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/color/LightModeTheme.ts:120]: image
image

**Review by jsartisan — APPROVED** [review]

File: app/client/packages/design-system/theming/src/color/LightModeTheme.ts (hunk 6 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/color/LightModeTheme.ts:120]: [Currently](https://github.com/appsmithorg/appsmith/blob/ba2c56c33d7dd08206cbfc3c510cb44d09ccbccf/app/client/packages/design-system/theming/src/color/ColorsAccessor.ts#L34) we're setting the range from `120` to `300` in OKLCh, looking at it more intently I realize that we may want to double-check it, similar to how we updated ranges for Green and Red from the starting proposal.
